x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings_by_resource/resource_findings/resource_findings_container.tsx

```diff
@@ -15,30 +15,30 @@ import * as TEST_SUBJECTS from '../../test_subjects';
 import { PageWrapper, PageTitle, PageTitleText } from '../../layout/findings_layout';
 import { useCspBreadcrumbs } from '../../../../common/navigation/use_csp_breadcrumbs';
 import { findingsNavigation } from '../../../../common/navigation/constants';
-import { useResourceFindings } from './use_resource_findings';
+import { ResourceFindingsQuery, useResourceFindings } from './use_resource_findings';
 import { useUrlQuery } from '../../../../common/hooks/use_url_query';
 import type { FindingsBaseURLQuery } from '../../types';
-import { getBaseQuery } from '../../utils';
+import { getBaseQuery, getPaginationQuery, getPaginationTableParams } from '../../utils';
 import { ResourceFindingsTable } from './resource_findings_table';
 import { FindingsSearchBar } from '../../layout/findings_search_bar';
 
-const getDefaultQuery = (): FindingsBaseURLQuery => ({
+const getDefaultQuery = (): FindingsBaseURLQuery & ResourceFindingsQuery => ({
   query: { language: 'kuery', query: '' },
   filters: [],
+  pageIndex: 0,
+  pageSize: 10,
 });
 
-const BackToResourcesButton = () => {
-  return (
-    <Link to={generatePath(findingsNavigation.findings_by_resource.path)}>
-      <EuiButtonEmpty iconType={'arrowLeft'}>
-        <FormattedMessage
-          id="xpack.csp.findings.resourceFindings.backToResourcesPageButtonLabel"
-          defaultMessage="Back to group by resource view"
-        />
-      </EuiButtonEmpty>
-    </Link>
-  );
-};
+const BackToResourcesButton = () => (
+  <Link to={generatePath(findingsNavigation.findings_by_resource.path)}>
+    <EuiButtonEmpty iconType={'arrowLeft'}>
+      <FormattedMessage
+        id="xpack.csp.findings.resourceFindings.backToResourcesPageButtonLabel"
+        defaultMessage="Back to group by resource view"
+      />
+    </EuiButtonEmpty>
+  </Link>
```

Comment:
do we actually need both `EuiButton` and react-router-dom `Link`? consider doing:
```
const BackToResourcesButton = () => {
  const { push } = useHistory();

  return (
    <EuiButtonEmpty
      iconType={'arrowLeft'}
      onClick={() => push(generatePath(findingsNavigation.findings_by_resource.path))}
    >
      <FormattedMessage
        id="xpack.csp.findings.resourceFindings.backToResourcesPageButtonLabel"
        defaultMessage="Back to group by resource view"
      />
    </EuiButtonEmpty>
  );
};
```
[Eui don't mind using buttons as links and the opposite aswell](https://elastic.github.io/eui/#/navigation/button#buttons-as-links)

is there a way to do a push with href?

Reply:
i agree it's unfortunate, but IMO better than using `onClick`, which makes it a non-anchor link. this means you can't:
- copy the link 
- `ctrl+click` the link to open in a new tab

what i'd like to do is use `href` in `EuiButtonEmpty` but it doesn't work with `react-router` 
there's some info on this [here](https://github.com/elastic/eui/blob/main/wiki/react-router.md) but i didn't manage to make anything out of it.